scripts/sanity_chk/sanitylib.py

```diff
@@ -3297,6 +3299,57 @@ def csv_report(self, filename):
                     rowdict["rom_size"] = rom_size
                 cw.writerow(rowdict)
 
+    def json_report(self, filename):
+        rowdict = {}
+        rowdict["environment"] = {"os": os.name,
+                                    "toolchain": self.get_toolchain()
+                                    }
+        rowdict["discards"] = []
```

Comment:
data from discard is now part of the overall report per test, so no need to have this as part of the output. when a test is skipped, there has to be a reason which is part of the instance.